8.16.6 release notes (#6650)
elastic/security-docs: docs/release-notes/8.16.asciidoc

[[release-notes-header-8.16.0]]
== 8.16

[discrete]
[[release-notes-8.16.6]]
=== 8.16.6

[discrete]
[[known-issue-8.16.6]]
==== Known issues

// tag::known-issue[]
[discrete]
.Duplicate alerts can be produced from manually running threshold rules 
[%collapsible]
====

> *Details* +

Reviewer: @nkhristinin can you take a quick look, please.

On November 12, 2024, it was discovered that manually running threshold rules could produce duplicate alerts if the date range was already covered by a scheduled rule execution.
====
// end::known-issue[]

// tag::known-issue[]
[discrete]
.Manually running custom query rules with suppression could suppress more alerts than expected
[%collapsible]
====
*Details* +
On November 12, 2024, it was discovered that manually running a custom query rule with suppression could incorrectly inflate the number of suppressed alerts. 
====
// end::known-issue[]

[discrete]
[[bug-fixes-8.16.6]]
==== Bug fixes

> * Fixes a bug that affected column widths in Timeline ({kibana-pull}214178[#214178]).

Reviewer 2: nit: Fix a bug that prevented column widths from being changed in Timeline

* Fixes a bug that prevented you from being able to save event filters ({kibana-pull}213805[#213805]).
* Allows Automatic Import to handle a variety of structures when parsing system logs ({kibana-pull}212611[#212611]).

> * Allows data view selections to persist when you save a Timeline ({kibana-pull}211343[#211343]).

Reviewer 2: Fixes a bug, preventing the selected data view / alerts only option from being reliably persisted in timeline

Author: ```suggestion
* Fixes a bug that prevented the selected data view and **Show only detection alerts** option from being reliably persisted in Timeline ({kibana-pull}211343[#211343]).
```

====
// end::known-issue[]

// tag::known-issue[]
[discrete]
.Manually running custom query rules with suppression could suppress more alerts than expected
[%collapsible]
====
*Details* +
On November 12, 2024, it was discovered that manually running a custom query rule with suppression could incorrectly inflate the number of suppressed alerts. 
====
// end::known-issue[]

[discrete]
[[bug-fixes-8.16.6]]
==== Bug fixes
* Fixes a bug that affected column widths in Timeline ({kibana-pull}214178[#214178]).
* Fixes a bug that prevented you from being able to save event filters ({kibana-pull}213805[#213805]).
* Allows Automatic Import to handle a variety of structures when parsing system logs ({kibana-pull}212611[#212611]).
* Allows data view selections to persist when you save a Timeline ({kibana-pull}211343[#211343]).
* Improves trusted applications performance by dropping Event Tracing for Windows (ETW) events earlier.
* Improves ETW-TI performance by not enabling local executable virtual memory events for sandboxed and Antimalware-PPL processes.

> * Fixes the low probability of Tamper Protected {elastic-defend} upgrade failure.

Reviewer 3: Idk if "tamper protected" should be cap? Please disregard if so. 
```suggestion
* Fixes the low probability of tamper-protected {elastic-defend} upgrade failure.
```

Author: According to these docs, it looks like it should be lower-case. Thanks, @jmikell821!

Author: Fixed in [0e5b413](https://github.com/elastic/security-docs/pull/6650/commits/0e5b413261aeb489eb360c169dd2aa23afb39457).

> * Improves trusted applications performance by dropping Event Tracing for Windows (ETW) events earlier.
> * Improves ETW-TI performance by not enabling local executable virtual memory events for sandboxed and Antimalware-PPL processes.
> * Fixes the low probability of Tamper Protected {elastic-defend} upgrade failure.
> * Fixes a bug in the scan response action that can crash {elastic-defend}.
> * Improves the `call_stack_final_user_module` and `call_stack_final_hook_module` logic.

Reviewer 4: ```suggestion
* Fixes a rare upgrade failure when {elastic-defend} has Tamper Protection enabled.
* Fixes a bug in the scan response action that can crash {elastic-defend}.
* Fixes a potential {elastic-defend} crash when generating multiple ransomware alerts on Windows. This issue was simultaneously mitigated by a cloud artifact update (manifest version 1.0.1381) on February 24, 2025. Internet-connected instances of {elastic-defend} will automatically receive this update -- no user intervention required. Air-gapped customers hosting their own artifacts should follow <<offline-endpoint,these instructions>>. We would like to acknowledge https://www.todyl.com[Todyl] for their assistance with this issue.
* Fixes a bug in {elastic-defend} for Linux where tty capture limit defaults were ignored.
```

Several of commits landed after 8.16.6 BC2, which is currently the final BC.  They won't ship in an 8.16.* release unless there's either another 8.16.6 BC or an 8.16.7 release.

Also, the latter two were merged into the `8.16` branch after the final 8.16.5 BC, so they're actually in 8.16.6.  I can't edit the 8.16.5 notes below from here though.

Author: I removed the last two PRs from the 8.16.5 release notes in [f5c3686](https://github.com/elastic/security-docs/pull/6650/commits/f5c3686077ca5a185f8e7de6ceae6ec4e5a80c10).
